entitlement/src/main/java/org/killbill/billing/entitlement/api/DefaultEntitlement.java

```diff
@@ -375,6 +375,55 @@ public Entitlement doCall(final EntitlementApi entitlementApi, final DefaultEnti
 
         return pluginExecution.executeWithPlugin(cancelEntitlementWithPlugin, pluginContext);
     }
+    
+
+    @Override
+    public Entitlement cancelEntitlementWithDate(final DateTime entitlementEffectiveDate, final DateTime billingEffectiveDate,
+                                                 final Iterable<PluginProperty> properties, final CallContext callContext) throws EntitlementApiException {
+        logCancelEntitlement(log, this, null, null, null, null, entitlementEffectiveDate, billingEffectiveDate); //TODO_1375 - commented for now to avoid compilation errors
+        checkForPermissions(Permission.ENTITLEMENT_CAN_CANCEL, callContext);
+        // Get the latest state from disk
+        refresh(callContext);
+        if (entitlementEffectiveDate != null && entitlementEffectiveDate.compareTo(getEffectiveStartDate()) < 0) {
+            throw new EntitlementApiException(ErrorCode.SUB_INVALID_REQUESTED_DATE, entitlementEffectiveDate, getEffectiveStartDate());
+        }
+        if (billingEffectiveDate != null && billingEffectiveDate.compareTo(getSubscriptionBase().getStartDate()) < 0) { //TODO_1375 - Added this check for billingEffectiveDate similar to the check above for entitlementEffectiveDate. Is the comparison to getSubscriptionBase().getStartDate() correct?
+            throw new EntitlementApiException(ErrorCode.SUB_INVALID_REQUESTED_DATE, entitlementEffectiveDate, getEffectiveStartDate());
+        }
+        //TODO_1375 - Is a null check required here for entitlementEffectiveDate and billingEffectiveDate?
```

Comment:
I would add one, in the `if` above, e.g.

```
if (entitlementEffectiveDate == null || (entitlementEffectiveDate != null && entitlementEffectiveDate.compareTo(getEffectiveStartDate()) < 0))
```

Reply:
Fixed